components/base/kinematicbase/execution.go

```diff
@@ -219,16 +220,16 @@ func (ptgk *ptgBaseKinematics) trajectoryArcSteps(
 
 	finalSteps := []arcStep{}
 	timeStep := 0.
-	curDist := 0.
-	curInputs := []referenceframe.Input{
+	curDist := inputs[startDistanceAlongTrajectoryIndex].Value
```

Comment:
isn't this still zero?, this is just a better way of referencing that value?

Reply:
No. This can be other values than zero, due to us evaluating partial trajectories left from prior course corrections.